tests/core/test_composite.py

```diff
@@ -468,7 +468,7 @@ def test_transform_filter(ant, sphere, airplane, tetbeam, inplace):
     keys_after = output.keys()
 
     assert (output is multi) == inplace
-    assert [(block_in is block_out) == inplace for block_in, block_out in zip(multi, output)]
+    assert all((block_in is block_out) == inplace for block_in, block_out in zip(multi, output))
```

Comment:
This test is failing. `MultiBlock.transform` will need to be updated. I explored using `copy` as part of the solution but there are issues with `copy`, see #6590

Reply:
Another issue with `__next__` was found which was causing this to fail by using `zip`. See #6600